File: catraca_virtual/src/main/java/br/edu/unilab/catraca/dao/TurnoDAO.java
package br.edu.unilab.catraca.dao;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

import br.edu.unilab.unicafe.model.Turno;

public class TurnoDAO extends DAO{
	public TurnoDAO(){
		super();
	}
	public TurnoDAO(int tipoDeConexao){
		super(tipoDeConexao);
		
	}
	public TurnoDAO(Connection conexao){
		super(conexao);
	}
	
	
	public boolean limpar(){
		
		PreparedStatement ps2;
		try {
			ps2 = this.getConexao().prepareStatement("DELETE FROM turno");
			ps2.executeUpdate();
			return true;
		} catch (SQLException e) {
			e.printStackTrace();
		}
		return false;
		
		
	}
	public void mostrar(){
		try {
			
			PreparedStatement ps = this.getConexao().prepareStatement("SELECT * FROM turno");
			ResultSet rs = ps.executeQuery();
			while(rs.next()){
				System.out.println("ID: "+rs.getString("turn_id"));
				System.out.println("Descricao: "+rs.getDouble("turn_descricao"));
				
				
			}
		} catch (SQLException e) {
			e.printStackTrace();
		}
		
	}
	public ArrayList<Turno>retornaLista(){
		try {
			ArrayList<Turno> lista = new ArrayList<Turno>();
			PreparedStatement ps = this.getConexao().prepareStatement("SELECT * FROM turno");
			ResultSet rs = ps.executeQuery();
			while(rs.next()){
				
				Turno turno = new Turno();
				turno.setId(rs.getInt("turn_id"));
				turno.setDescricao(rs.getString("turn_descricao"));
				turno.setHoraInicial(rs.getString("turn_hora_inicio"));
				turno.setHoraFinal(rs.getString("turn_hora_fim"));
				lista.add(turno);
				
				
			}
			return lista;
		} catch (SQLException e) {
			e.printStackTrace();
		}
		return null;
		
	}
	
	
	public boolean inserir(Turno turno){
		try {
			
			PreparedStatement ps2 = this.getConexao().prepareStatement("INSERT INTO turno(turn_id, turn_hora_inicio, turn_hora_fim, turn_descricao) VALUES(?, ?, ?, ?)");
			
			ps2.setInt(1, turno.getId());
			ps2.setString(2, turno.getHoraInicial());
			ps2.setString(3, turno.getHoraFinal());
			ps2.setString(4, turno.getDescricao());
			
			ps2.executeUpdate();
			return true;
					
			
		} catch (SQLException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return false;
		}		
		
		
	}
	
	

}

File: web/classes/dao/TurnoDAO.php
<?php

/**
 * Classe utilizada para conxão com o Bando de Dados.
 * @author Jefferson Uchoa Ponte
 * @version 1.0
 * @copyright UNILAB - Universidade da Integracao Internacional da Lusofonia Afro-Brasileira.
 * @package DAO
 */

/**
 * UnidadeDAO alterações do banco de dados referentes à entidade unidade.
 *
 * Gera pesistencia da classe unidade.
 * a unidade academica.
 * 
 * @author jefponte
 *        
 */
class TurnoDAO extends DAO {
	
	/**
	 * Retorna Um vetor de unidades.
	 *
	 * @return Turno[]
	 */
	public function retornaLista() {
		$lista = array ();
		$result = $this->getConexao ()->query ( "SELECT * FROM turno" );
		
		foreach ( $result as $linha ) {
			$turno = new Turno ();
			$turno->setId ( $linha ['turn_id'] );
			$turno->setDescricao ( $linha ['turn_descricao'] );
			$turno->setHoraFinal ( $linha ['turn_hora_fim'] );
			$turno->setHoraInicial ( $linha ['turn_hora_inicio'] );
			$lista [] = $turno;
		}
		return $lista;
	}
	
	/**
	 * Função Utilizada para Inserir um turno. 
	 * @param Turno $turno Objeto Turno contendo os dados do turno.
	 */
	public function inserir(Turno $turno) {
		$descricao = $turno->getDescricao ();
		$horaInicial = $turno->getHoraInicial ();
		$horaFinal = $turno->getHoraFinal ();
		
		if ($this->getConexao ()->query ( "INSERT INTO turno (turn_hora_inicio, turn_hora_fim, turn_descricao) VALUES('$horaInicial', '$horaFinal','$descricao')" ))
			return true;
		return false;
	}
	
	/**
	 * Realiza um pesquisa pelo Id do turno.
	 * @param Turno $turno
	 */
	public function retornaTurnoPorId(Turno $turno) {
		$idTurno = $turno->getId ();
		$sql = "SELECT * FROM turno WHERE turn_id = $idTurno";
		$result = $this->getConexao ()->query ( $sql );
		foreach ( $result as $linha ) {
			$turno->setId ( $linha ['turn_id'] );
			$turno->setHoraInicial ( $linha ['turn_hora_inicio'] );
			$turno->setHoraFinal ( $linha ['turn_hora_fim'] );
			$turno->setDescricao ( $linha ['turn_descricao'] );
		}
	}
	
	/**
	 * Realiza a atualização do 
	 * @param Turno $turno
	 */
	public function atualizaTurno(Turno $turno) {
		$idTurno = $turno->getId ();
		$novaHoraIni = $turno->getHoraInicial ();
		$novaHoraFim = $turno->getHoraFinal ();
		
		$sql = "UPDATE turno SET
		turn_hora_inicio = '$novaHoraIni', turn_hora_fim = '$novaHoraFim'
		WHERE turn_id = $idTurno";
		if ($this->getConexao ()->exec ( $sql ))
			return true;
		return false;
	}
}

?>